environment.h

```diff
@@ -103,6 +109,9 @@ class environment
     ///The environmental function that is currently used to determine the optimal output
     std::function<double(std::vector<double>)> m_current_function;
 
+    /// The name of the current function
+    char m_name_current_function;
```

Comment:
Yes this is the suggestion I had in mind more or less. I would have used to `bool` like `m_curr_func_is_A` `m_curr_func_is_B`, but this also works; although the name might accidentaly switch to another character that is neither A or B, but there are ways to control that. 
An even simpler solution is to have only one `bool` `m_curr_func_is_A`.